internal/internal_task_pollers.go

```diff
@@ -925,32 +912,22 @@ func reportActivityComplete(ctx context.Context, service workflowservice.Workflo
 	var reportErr error
 	switch request := request.(type) {
 	case *workflowservice.RespondActivityTaskCanceledRequest:
-		reportErr = backoff.Retry(ctx,
-			func() error {
-				grpcCtx, cancel := newGRPCContext(ctx, grpcMetricsScope(rpcScope))
-				defer cancel()
-
-				_, err := service.RespondActivityTaskCanceled(grpcCtx, request)
-				return err
-			}, createDynamicServiceRetryPolicy(ctx), isServiceTransientError)
+		grpcCtx, cancel := newGRPCContext(ctx, grpcMetricsScope(rpcScope),
```

Comment:
This is beautiful! What blocks you from removing `isServiceTransientError` func? We have this logic in two places now.

Reply:
There are 2 more places that require a little bit of thinking. Let me try to address them too.

Reply:
There were couple of places with custom error handling, that I left for later. Let me try clean them up too.